server/models/procedures/invoicing.sql

```diff
@@ -771,3 +771,79 @@ BEGIN
       HAVING balance <> 0
     ) AS zz ORDER BY zz.date;
 END $$
+
+
+-- This Procedure retrieve the balace(debit, credit) of each invoices during a specified period
+
+DROP PROCEDURE IF EXISTS `UnbalancedInvoicePayments`$$
+CREATE   PROCEDURE `UnbalancedInvoicePayments`(IN dateFrom DATE, IN dateTo DATE)
+BEGIN
+  DECLARE _uuid, _debtor_uuid, _tempRef BINARY(16);
+  DECLARE _invoiceID VARCHAR(100);
+  DECLARE _date DATE;
+
+  DECLARE done BOOLEAN;
+  DECLARE curs1 CURSOR FOR 
+    SELECT  i.uuid, i.debtor_uuid, i.date 
+    FROM invoice i
+    WHERE i.date BETWEEN dateFrom AND dateTo
+    ORDER BY DATE;
+   
+    DECLARE CONTINUE HANDLER FOR NOT FOUND SET done = TRUE;
+
+  -- all invoice balances will be stored temporarely in stage_invoice_balance table
+
+  CREATE TEMPORARY TABLE IF NOT EXISTS stage_invoice_balance(
+    tempRef BINARY(16),
+    invoice_uuid BINARY(16),
+    debit DECIMAL(19, 4),
+    credit DECIMAL(19, 4),
+    debtor_uuid BINARY(16),
+    DATE DATE,
+    reference VARCHAR(100)
+  );
+
+  SET _tempRef = HUID(UUID());
+
+  OPEN curs1;
+    read_loop: LOOP
+    FETCH curs1 INTO _uuid, _debtor_uuid, _date;
+      IF done THEN
+        LEAVE read_loop;
+      END IF;
+      -- some payment may be posted in the general ledger so it union  with posting_journal is required
+
+      SELECT IFNULL(d.text, '') INTO _invoiceID 
+      FROM document_map d WHERE d.uuid = _uuid
+      LIMIT 1;
+      -- check if the invoice is debited in stage_invoice_balance, this operation should not be done twise
+      -- the invoice must be debited once and after
+
+      INSERT INTO stage_invoice_balance
+      SELECT _tempRef, _uuid, SUM(invoices_payments.debit_equiv) AS debit, SUM(invoices_payments.credit_equiv) AS credit, _debtor_uuid, _date, _invoiceID
+      FROM
+	      ( SELECT  p.record_uuid AS UUID, p.debit_equiv, p.credit_equiv
+          FROM posting_journal p
+          WHERE p.record_uuid = _uuid AND p.entity_uuid = _debtor_uuid
+          UNION ALL
+          SELECT p.record_uuid AS UUID, p.debit_equiv, p.credit_equiv
+          FROM general_ledger p
+          WHERE  (p.record_uuid = _uuid AND p.entity_uuid = _debtor_uuid) OR (p.reference_uuid = _uuid AND p.entity_uuid = _debtor_uuid)
+        ) AS invoices_payments
+      GROUP BY invoices_payments.uuid;
+
+    END LOOP;
+  CLOSE curs1;
+
+  SELECT x.invoice_uuid, x.debtor_uuid, x.debit, x.credit, (x.debit - x.credit) AS balance,
```

Comment:
It would be interesting to have more statistics on this.  You can easily create a lightweight temporary table to handle this case.

```sql
CREATE TEMPORARY TABLE `invoice_balances` AS
  SELECT x.invoice_uuid, x.debtor_uuid, x.debit, x.credit, (x.debit - x.credit) AS balance,
   (x.credit/x.debit) AS paymentPercentage , x.reference
  FROM (
    SELECT  BUID(invoice_uuid) AS invoice_uuid, BUID(debtor_uuid) AS debtor_uuid, 
    SUM(debit) AS debit, SUM(credit) AS credit, DATE AS creation_date, reference
    FROM stage_invoice_balance
    WHERE tempRef = _tempRef
    GROUP BY reference
  ) AS x
  WHERE  x.debit <> x.credit;
```

Now you can do things like this:
```sql
SET totalInvoiceValue = (SELECT SUM(debit) FROM invoice_balances);
SELECT
  SUM(IF(balance / totalInvoiceValue < 0.25, 1, 0) as _0_25,
  SUM(IF(balance / totalInvoiceValue >= 0.25 AND (balance / totalInvoiceValue) < 0.50, 1, 0) AS _25_50,
  SUM(IF(balance / totalInvoiceValue >= 0.50 AND (balance / totalInvoiceValue) < 0.75, 1, 0) AS _50_75,
  SUM(IF(balance / totalInvoiceValue >= 0.75 AND (balance / totalInvoiceValue) < 1, 1, 0) AS _75_100,
  SUM(IF(balance / totalInvoiceValue > 100, 1, 0) AS 100_plus,
FROM invoice_balances;
```
to generate any statistics you need.